File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int answer = 0;
        while(storey != 0) {
            int mod = storey % 10;
            if(mod < 5) {
                answer += mod;
            }
            else if(mod > 5) {
                answer += 10 - mod;
                storey += 10- mod;
            }
            else {
                int prev = storey / 10 % 10;
                if(prev < 5) {
                    answer +=5;
                }
                else {
                    answer += 5;
                    storey += 5;
                }
            }
           
            storey /= 10;
        }
        
        
     
        return answer;
    }
}

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    var answer = 0;
    while (storey > 0) {
        const cur = storey % 10;
        const next = Math.floor(storey / 10) % 10;
        if (cur > 5 || (cur === 5 && next >= 5)) {
            answer += (10 - cur);
            storey += 10;
        } else answer += cur;
        storey = Math.floor(storey/10);
    
    }
    return answer;
}